cpp/tests/test_math_time_series_functor.cpp

```diff
@@ -0,0 +1,80 @@
+#include "memilio/math/time_series_functor.h"
+#include "memilio/utils/random_number_generator.h"
+
+#include <gtest/gtest.h>
+
+class TestMathTimeSeriesFunctor : public ::testing::Test
+{
+public:
+    using TSF = mio::TimeSeriesFunctor<double>;
+
+    const int num_evals = 1000;
+    const double min = -1e+3, max = 1e+3; // a reasonably large range for fuzzy_val
+
+    double fuzzy_val(double min_, double max_)
+    {
+        return mio::UniformDistribution<double>::get_instance()(m_rng, min_, max_);
+    }
+
+protected:
+    void SetUp() override
+    {
+        log_rng_seeds(m_rng, mio::LogLevel::warn);
+    }
+
+private:
+    mio::RandomNumberGenerator m_rng{};
+};
+
+TEST_F(TestMathTimeSeriesFunctor, zero)
+{
+    // Test that the Zero-functor always returns zero, using a random evaluation point.
+
+    // initialize functor using the default ctor
+    TSF tsf;
+
+    // check one deterministic value first to avoid flooding the test output with failed tests
+    ASSERT_EQ(tsf(0.0), 0.0);
+    // verify output
+    for (int i = 0; i < this->num_evals; i++) {
+        auto random_t_eval = this->fuzzy_val(this->min, this->max);
+        EXPECT_EQ(tsf(random_t_eval), 0.0);
+    }
+}
+
+TEST_F(TestMathTimeSeriesFunctor, linearInterpolation)
```

Comment:
I think having the random tests is OK, but a test with fixed values would be more readable, e.g., interpolate a time series of "nice" values at "nice" time points, for some definition of nice, and compare with fixed values that are easily checked with mental arithmetic. Could you add at least one such case?